oracle/oracle.go

```diff
@@ -197,6 +208,52 @@ func (or *Oracle) AdvanceStateToNextSlot(fullBlock *FullBlock) (uint64, error) {
 	return processedSlot, nil
 }
 
+// Unsubscribes validators that are not active in ethereum. Shares their pending rewards to the pool
+// TODO: there are multiple validator statuses that should be considered as "not active" and should be unsubscribed.
+func (or *Oracle) ValidatorCleanup(blockHeader *v1.BeaconBlockHeader) error {
+	or.mutex.Lock()
+	defer or.mutex.Unlock()
+
+	// Only cleanup if we're past the cleanup slot fork
+	if uint64(blockHeader.Header.Message.Slot) >= MainnetCleanupValidatorsSlot {
```

Comment:
We need a logic here to handle both mainnet and testnet. similar to the other PR.

var RewardSlotFork = map[string]uint64{
	"mainnet": MainnetRewardsSlotFork,
	"holesky": HoleskyRewardsSlotFork,
}

I can take care of this in the other PR, and then you can rebase and take it from it.